backend/main/server/resources/Video.py

```diff
@@ -2,11 +2,28 @@
 from flask import request
 from main.server import db, cache, app
 from main.server.models import Video, VideoSchema
-from flask_jwt import jwt_required
 
 video_schema = VideoSchema()
 videos_schema = VideoSchema(many=True)
 
+def insertVideo(col, data):
+    videoLink = artistLink = username = title = None
+    for i in range(0, len(data)):
+        if col[i] == "videoLink" and data[i]: videoLink = data[i]
+        elif col[i] == "artistLink": artistLink = data[i]
+        elif col[i] == "username": username = data[i]
+        elif col[i] == "title": title = data[i]
+        else: continue
+    message = Video.query.filter_by(
+            videoLink=videoLink).first()
+    if message:
+        return 1
+    message = Video(videoLink=videoLink,
+                    artistLink=artistLink,
+                    username=username,
+                    title=title)
+    db.session.add(message)
```

Comment:
I'm not sure the right thing to do here is to add everything to the session and expect the downstream ops to commit. I would put the commit here, and pass the dry_run arg here and skip db add and commit.

Reply:
It would be no different than what we're doing today except that we're making calls through an HTTP request. Also, looks like commits auto-flush anyways https://docs.sqlalchemy.org/en/13/orm/session_basics.html#committing

Reply:
Reading the docs shows that:
Session.commit() is used to commit the current transaction. It always issues Session.flush() beforehand to flush any remaining state to the database; this is independent of the “autoflush” setting

Also I'm pretty sure that the reason our old API endpoint is so slow is because it keeps flushing data to the disk over and over and over again for each entry.

I think we are probably gonna leave the commit in the main part of the script because then the db isn't cluttered by a faulty csv